deltacat/compute/compactor/steps/materialize.py

```diff
@@ -101,69 +149,60 @@ def materialize(
             src_file_idx_np.item(),
             file_reader_kwargs=read_kwargs_provider,
         )
-        mask_pylist = list(repeat(False, len(pa_table)))
+        record_count = len(pa_table)
+        if record_count > max_records_per_output_file:
```

Comment:
I think we'll want to get rid of this in a follow-up PR since some regressive tables will present 1 or more overly large files from amongst their deltas (meaning that their file size is a bottleneck for bulk parallel consumption) and our job in general should be to always produce a read-optimized view of the table by the end of materialize by not only merging small files up to `max_records_per_output_file`, but also splitting large files down to `max_records_per_output_file`.